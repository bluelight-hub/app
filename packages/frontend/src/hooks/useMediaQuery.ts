import { useEffect, useState } from 'react';

/**
 * Hook to check if a media query matches the current viewport
 * @param query Media query string (e.g. '(max-width: 480px)')
 * @returns Boolean indicating if the query matches
 */
export const useMediaQuery = (query: string): boolean => {
    const [matches, setMatches] = useState<boolean>(false);

    useEffect(() => {
        // Create media query list
        const mediaQuery = window.matchMedia(query);

        // Set initial value
        setMatches(mediaQuery.matches);

        // Define callback for changes
        const handleResize = (event: MediaQueryListEvent) => {
            setMatches(event.matches);
        };

        // Add event listener
        mediaQuery.addEventListener('change', handleResize);

        // Cleanup
        return () => {
            mediaQuery.removeEventListener('change', handleResize);
        };
    }, [query]);

    return matches;
};

export default useMediaQuery; 