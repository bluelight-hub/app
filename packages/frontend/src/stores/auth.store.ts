import { Store } from '@tanstack/react-store';
import type { UserResponseDto } from '@bluelight-hub/shared/client';

/**
 * Interface für den Authentifizierungs-Store
 */
export interface AuthState {
  /** Der aktuell authentifizierte Benutzer oder null wenn nicht angemeldet */
  user: UserResponseDto | null;
  /** Ob der Benutzer authentifiziert ist */
  isAuthenticated: boolean;
  /** Ob gerade eine Authentifizierung läuft */
  isLoading: boolean;
  /** Ob der Benutzer als Admin authentifiziert ist */
  isAdminAuthenticated: boolean;
}

/**
 * TanStack Store für die Authentifizierung
 *
 * Verwaltet den globalen Authentifizierungszustand der Anwendung.
 * Der Store ist framework-agnostisch und bietet fine-grained reactivity,
 * sodass Komponenten nur bei Änderungen der abonnierten Properties re-rendern.
 */
export const authStore = new Store<AuthState>({
  user: null,
  isAuthenticated: false,
  isLoading: false,
  isAdminAuthenticated: false,
});

/**
 * Actions für den Auth Store
 *
 * Kapselt alle Authentifizierungs-bezogenen Aktionen
 */
export const authActions = {
  /**
   * Setzt den authentifizierten Benutzer
   *
   * @param user - Die Benutzerdaten oder null zum Ausloggen
   */
  setUser: (user: UserResponseDto | null) => {
    authStore.setState((state) => ({
      ...state,
      user,
      isAuthenticated: !!user,
    }));
  },

  /**
   * Setzt den Loading-Status
   *
   * @param isLoading - Der neue Loading-Status
   */
  setLoading: (isLoading: boolean) => {
    authStore.setState((state) => ({
      ...state,
      isLoading,
    }));
  },

  /**
   * Meldet den Benutzer ab und löscht den Store
   */
  logout: () => {
    authStore.setState(() => ({
      user: null,
      isAuthenticated: false,
      isLoading: false,
      isAdminAuthenticated: false,
    }));
  },

  /**
   * Hilfsfunktion für Login/Registrierung
   * Setzt User und beendet Loading-Status
   *
   * @param user - Die Benutzerdaten nach erfolgreicher Authentifizierung
   */
  loginSuccess: (user: UserResponseDto) => {
    authStore.setState((state) => ({
      ...state,
      user,
      isAuthenticated: true,
      isLoading: false,
    }));
  },

  /**
   * Setzt den Admin-Authentifizierungsstatus
   *
   * @param isAdminAuthenticated - Ob der Benutzer als Admin authentifiziert ist
   */
  setAdminAuth: (isAdminAuthenticated: boolean) => {
    authStore.setState((state) => ({
      ...state,
      isAdminAuthenticated,
    }));
  },
};
