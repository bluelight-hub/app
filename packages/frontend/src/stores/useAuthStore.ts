import { create } from 'zustand';
import { devtools, persist } from 'zustand/middleware';
import { api } from '../api';
import { logger } from '../utils/logger';
import { authStorage } from '../utils/authStorage';
import { AuthUserDto } from '@bluelight-hub/shared/client';

interface AuthState {
  user: AuthUserDto | null;
  isAuthenticated: boolean;
  isLoading: boolean;
  isRefreshing: boolean;
  accessToken: string | null;
  refreshToken: string | null;
  sessionExpiresAt: Date | null;
  lastRefreshAt: Date | null;
  refreshTimer: NodeJS.Timeout | null;
}

interface AuthActions {
  setUser: (user: AuthUserDto | null) => void;
  setTokens: (accessToken: string, refreshToken?: string, expiresIn?: number) => Promise<void>;
  login: (email: string, password: string) => Promise<{ success: boolean; error?: string }>;
  logout: () => Promise<void>;
  refreshAccessToken: () => Promise<boolean>;
  checkAuthStatus: () => Promise<void>;
  hasRole: (role: string) => boolean;
  hasPermission: (permission: string) => boolean;
  isAdmin: () => boolean;
  scheduleTokenRefresh: () => void;
  clearRefreshTimer: () => void;
}

export interface AuthStore extends AuthState, AuthActions {}

const REFRESH_BUFFER_SECONDS = 60; // Refresh 1 minute before expiry
const MIN_REFRESH_INTERVAL = 30000; // Minimum 30 seconds between refresh attempts

export const useAuthStore = create<AuthStore>()(
  devtools(
    persist(
      (set, get) => ({
        // Initial state
        user: null,
        isAuthenticated: false,
        isLoading: false,
        isRefreshing: false,
        accessToken: null,
        refreshToken: null,
        sessionExpiresAt: null,
        lastRefreshAt: null,
        refreshTimer: null,

        // Actions
        setUser: (user) => set({ user, isAuthenticated: !!user }),

        setTokens: async (accessToken, refreshToken, expiresIn) => {
          // Clear existing refresh timer
          get().clearRefreshTimer();

          // Calculate session expiry
          const sessionExpiresAt = expiresIn
            ? new Date(Date.now() + expiresIn * 1000)
            : new Date(Date.now() + 15 * 60 * 1000); // Default 15 minutes

          // Store tokens
          await authStorage.setTokens(accessToken, refreshToken);

          set({
            accessToken,
            refreshToken: refreshToken || get().refreshToken,
            sessionExpiresAt,
            lastRefreshAt: new Date(),
          });

          // Schedule next refresh
          get().scheduleTokenRefresh();

          logger.debug('Tokens updated', {
            expiresAt: sessionExpiresAt.toISOString(),
            hasRefreshToken: !!refreshToken,
          });
        },

        login: async (email, password) => {
          set({ isLoading: true });

          try {
            const response = await api.auth.authControllerLoginV1Raw({
              loginDto: { email, password },
            });
            const data = await response.value();

            if (data) {
              logger.info('Login successful', { email });

              // Set user and tokens
              if (data.user) {
                set({ user: data.user, isAuthenticated: true });
              }

              if (data.accessToken) {
                await get().setTokens(data.accessToken, data.refreshToken, data.expiresIn);
              }

              set({ isLoading: false });
              return { success: true };
            } else {
              set({ isLoading: false });
              return {
                success: false,
                error: 'No data received from server',
              };
            }
          } catch (error) {
            logger.error('Login error', { error });
            set({ isLoading: false });
            return {
              success: false,
              error: error instanceof Error ? error.message : 'Network error',
            };
          }
        },

        logout: async () => {
          try {
            // Clear refresh timer first
            get().clearRefreshTimer();

            // Call logout endpoint to revoke session
            await api.auth.authControllerLogoutV1();
            logger.info('Logout successful');
          } catch (error) {
            logger.error('Logout error', { error });
          } finally {
            // Clear local state and storage
            set({
              user: null,
              isAuthenticated: false,
              accessToken: null,
              refreshToken: null,
              sessionExpiresAt: null,
              lastRefreshAt: null,
            });
            await authStorage.clearTokens();
          }
        },

        refreshAccessToken: async () => {
          const state = get();

          // Prevent concurrent refresh attempts
          if (state.isRefreshing) {
            logger.debug('Refresh already in progress');
            return false;
          }

          // Check minimum interval between refresh attempts
          if (state.lastRefreshAt) {
            const timeSinceLastRefresh = Date.now() - state.lastRefreshAt.getTime();
            if (timeSinceLastRefresh < MIN_REFRESH_INTERVAL) {
              logger.debug('Skipping refresh - too soon since last attempt', {
                timeSinceLastRefresh,
              });
              return false;
            }
          }

          const { refreshToken } = state;
          if (!refreshToken) {
            logger.warn('No refresh token available');
            return false;
          }

          set({ isRefreshing: true });

          try {
            const response = await api.auth.authControllerRefreshTokenV1Raw({
              refreshTokenDto: { refreshToken },
            });

            const data = await response.value();
            if (data) {
              logger.info('Token refresh successful');

              // Update tokens and user data
              if (data.accessToken) {
                await get().setTokens(data.accessToken, data.refreshToken, data.expiresIn);
              }

              if (data.user) {
                set({ user: data.user });
              }

              set({ isRefreshing: false });
              return true;
            } else {
              set({ isRefreshing: false });
              return false;
            }
          } catch (error: any) {
            logger.error('Token refresh error', { error });

            // If refresh fails with 401, clear auth state
            if (error?.response?.status === 401 || error?.status === 401) {
              await get().logout();
            }

            set({ isRefreshing: false });
            return false;
          }
        },

        checkAuthStatus: async () => {
          set({ isLoading: true });

          try {
            const hasTokens = await authStorage.hasValidTokens();
            if (!hasTokens) {
              set({ isLoading: false });
              return;
            }

            // Load tokens from storage
            const tokens = await authStorage.getTokens();
            if (tokens.authToken) {
              set({
                accessToken: tokens.authToken,
                refreshToken: tokens.refreshToken || null,
              });
            }

            // Try to get current user
            try {
              const response = await api.auth.authControllerGetCurrentUserV1Raw();
              const userData = await response.value();
              if (userData) {
                set({
                  user: userData,
                  isAuthenticated: true,
                });
                logger.debug('User data loaded successfully');

                // Schedule token refresh
                get().scheduleTokenRefresh();
              } else {
                logger.warn('Failed to fetch user data');
                await get().logout();
              }
            } catch (error) {
              logger.error('Error fetching user data:', error);
              // Try to refresh token
              const refreshed = await get().refreshAccessToken();
              if (!refreshed) {
                await get().logout();
              }
            }
          } catch (error) {
            logger.error('Error checking auth status:', error);
          } finally {
            set({ isLoading: false });
          }
        },

        hasRole: (role) => {
          const { user } = get();
          return user?.roles?.includes(role) || false;
        },

        hasPermission: (permission) => {
          const { user } = get();
          return user?.permissions?.includes(permission) || false;
        },

        isAdmin: () => {
          const { user } = get();
          return (
            user?.roles?.some((role) => ['SUPER_ADMIN', 'ADMIN', 'SUPPORT'].includes(role)) || false
          );
        },

        scheduleTokenRefresh: () => {
          const state = get();

          // Clear any existing timer
          get().clearRefreshTimer();

          if (!state.sessionExpiresAt || !state.refreshToken) {
            logger.debug('Cannot schedule refresh - missing expiry or refresh token');
            return;
          }

          const now = Date.now();
          const expiresAt = state.sessionExpiresAt.getTime();
          const refreshAt = expiresAt - REFRESH_BUFFER_SECONDS * 1000;
          const delay = Math.max(refreshAt - now, 0);

          if (delay > 0) {
            logger.debug('Scheduling token refresh', {
              expiresAt: state.sessionExpiresAt.toISOString(),
              refreshIn: Math.round(delay / 1000) + 's',
            });

            const timer = setTimeout(async () => {
              logger.debug('Executing scheduled token refresh');
              await get().refreshAccessToken();
            }, delay);

            set({ refreshTimer: timer });
          } else {
            // Token already expired or about to expire, refresh immediately
            logger.debug('Token expired or about to expire, refreshing immediately');
            get().refreshAccessToken();
          }
        },

        clearRefreshTimer: () => {
          const { refreshTimer } = get();
          if (refreshTimer) {
            clearTimeout(refreshTimer);
            set({ refreshTimer: null });
          }
        },
      }),
      {
        name: 'auth-storage',
        partialize: (state) => ({
          // Only persist non-sensitive data
          user: state.user,
          isAuthenticated: state.isAuthenticated,
        }),
      },
    ),
    {
      name: 'AuthStore',
    },
  ),
);

// Selector hooks for common use cases
export const useUser = () => useAuthStore((state) => state.user);
export const useIsAuthenticated = () => useAuthStore((state) => state.isAuthenticated);
export const useIsLoading = () => useAuthStore((state) => state.isLoading);
export const useAuthActions = () =>
  useAuthStore((state) => ({
    login: state.login,
    logout: state.logout,
    hasRole: state.hasRole,
    hasPermission: state.hasPermission,
    isAdmin: state.isAdmin,
  }));
