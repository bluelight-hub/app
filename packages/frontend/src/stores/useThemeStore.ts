import { create } from 'zustand';
import { devtools, persist } from 'zustand/middleware';

interface ThemeState {
    dark: boolean;
    auto: boolean;
    // Für manuelle Änderungen durch den Benutzer
    setManualDark: (dark: boolean) => void;
    // Für automatische Änderungen durch das System
    setSystemDark: (dark: boolean) => void;
    setAuto: (auto: boolean) => void;
}

// Get initial system dark mode preference
const getInitialDarkMode = () => {
    if (typeof window === 'undefined') return false;
    return window.matchMedia('(prefers-color-scheme: dark)').matches;
};

export const useThemeStore = create<ThemeState>()(
    devtools(
        persist(
            (set) => {
                // Initialize with system preference
                const systemDark = getInitialDarkMode();

                return {
                    dark: systemDark,
                    auto: true,
                    // Manuelle Änderung durch den Benutzer
                    setManualDark: (dark) =>
                        set(() => ({
                            dark,
                            auto: false, // Automatik wird ausgeschaltet
                        })),
                    // Automatische Änderung durch das System
                    setSystemDark: (dark) =>
                        set((state) => ({
                            dark: state.auto ? dark : state.dark, // Nur ändern wenn auto aktiv
                            auto: state.auto, // Auto-Status bleibt unverändert
                        })),
                    setAuto: (auto) =>
                        set((state) => ({
                            auto,
                            dark: auto ? getInitialDarkMode() : state.dark,
                        })),
                };
            },
            {
                name: 'theme-storage',
            }
        )
    )
); 