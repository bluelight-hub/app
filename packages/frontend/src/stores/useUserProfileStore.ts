import { create } from 'zustand';
import { devtools } from 'zustand/middleware';

interface UserProfile {
    name: string;
    imageUrl: string;
    email?: string;
}

interface UserProfileState {
    profile: UserProfile | null;
    isLoading: boolean;
    error: string | null;
    setProfile: (profile: UserProfile) => void;
    updateProfile: (updates: Partial<UserProfile>) => void;
    clearProfile: () => void;
}

export const useUserProfileStore = create<UserProfileState>()(
    devtools(
        (set) => ({
            profile: {
                name: 'Robert Retter',
                imageUrl: 'https://randomuser.me/api/portraits/women/11.jpg',
            },
            isLoading: false,
            error: null,
            setProfile: (profile) => set({ profile, error: null }),
            updateProfile: (updates) =>
                set((state) => ({
                    profile: state.profile ? { ...state.profile, ...updates } : null,
                    error: null,
                })),
            clearProfile: () => set({ profile: null, error: null }),
        }),
        { name: 'user-profile-store' }
    )
); 