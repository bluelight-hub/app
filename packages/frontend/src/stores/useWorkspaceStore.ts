import { create, StateCreator } from 'zustand';
import { devtools } from 'zustand/middleware';

export interface WorkspaceTab {
    id: string;
    title: string;
    type: 'empty' | 'project' | 'settings';
    active: boolean;
}

interface WorkspaceState {
    tabs: WorkspaceTab[];
    activeTabId: string | null;
    addTab: (type: WorkspaceTab['type'], title?: string) => void;
    removeTab: (id: string) => void;
    setActiveTab: (id: string) => void;
}

type WorkspaceStore = StateCreator<
    WorkspaceState,
    [["zustand/devtools", never]],
    [],
    WorkspaceState
>;

export const useWorkspaceStore = create<WorkspaceState>()(
    devtools(
        ((set) => ({
            tabs: [],
            activeTabId: null,
            addTab: (type: WorkspaceTab['type'], title?: string) => set((state: WorkspaceState) => {
                const newTab: WorkspaceTab = {
                    id: crypto.randomUUID(),
                    title: title || `New ${type} ${state.tabs.length + 1}`,
                    type,
                    active: true,
                };

                return {
                    tabs: [...state.tabs.map(tab => ({ ...tab, active: false })), newTab],
                    activeTabId: newTab.id,
                };
            }),
            removeTab: (id: string) => set((state: WorkspaceState) => {
                const newTabs = state.tabs.filter(tab => tab.id !== id);
                const newActiveTabId = state.activeTabId === id
                    ? newTabs[newTabs.length - 1]?.id || null
                    : state.activeTabId;

                return {
                    tabs: newTabs,
                    activeTabId: newActiveTabId,
                };
            }),
            setActiveTab: (id: string) => set((state: WorkspaceState) => ({
                tabs: state.tabs.map(tab => ({
                    ...tab,
                    active: tab.id === id,
                })),
                activeTabId: id,
            })),
        })) as WorkspaceStore,
        { name: 'workspace-store' }
    )
); 