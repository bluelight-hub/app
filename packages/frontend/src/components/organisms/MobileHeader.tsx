import { Button } from 'antd';
import React from 'react';
import { PiTextIndent } from 'react-icons/pi';
import useMediaQuery from '../../hooks/useMediaQuery';
import { useUserProfileStore } from '../../stores/useUserProfileStore';
import { StatusIndicator } from '../atoms/StatusIndicator';
import UserProfile from '../atoms/UserProfile';

interface MobileHeaderProps {
    title: string;
    onOpenSidebar: () => void;
}

/**
 * Mobile header component with menu button and user profile
 */
const MobileHeader: React.FC<MobileHeaderProps> = ({
    title,
    onOpenSidebar,
}) => {
    const profile = useUserProfileStore((state) => state.profile);
    const isSmallerScreen = useMediaQuery('(max-width: 480px)');

    if (!profile) return null;

    return (
        <div className="sticky top-0 z-40 flex items-center gap-x-6 bg-white dark:bg-gray-900 px-4 py-4 shadow-sm dark:shadow-gray-800 sm:px-6 lg:hidden">
            <Button
                type="text"
                className="-m-2.5 p-2.5 text-gray-700 dark:text-gray-300 lg:hidden"
                onClick={onOpenSidebar}
            >
                <span className="sr-only">Sidebar Ã¶ffnen</span>
                <PiTextIndent size={24} />
            </Button>
            <div className="flex-1 text-sm/6 font-semibold text-gray-900 dark:text-gray-100">
                <div className="flex items-center gap-2">
                    {title}
                    <StatusIndicator />
                </div>
            </div>
            <div>
                <UserProfile href="#" hideText={isSmallerScreen} />
            </div>
        </div>
    );
};

export default MobileHeader; 