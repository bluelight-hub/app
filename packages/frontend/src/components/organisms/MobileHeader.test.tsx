import { fireEvent, render, screen } from '@testing-library/react';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { useUserProfileStore } from '../../stores/useUserProfileStore';
import MobileHeader from './MobileHeader';

// Definiere die benötigten Typen lokal, da sie nicht exportiert werden
interface UserProfile {
    name: string;
    imageUrl: string;
    email?: string;
}

interface MockUserProfileState {
    profile: UserProfile | null;
    isLoading: boolean;
    error: string | null;
    setProfile: (profile: UserProfile) => void;
    updateProfile: (updates: Partial<UserProfile>) => void;
    clearProfile: () => void;
}

// Mock für den Store mit verschiedenen Implementierungen
const mockProfile = {
    name: 'Test User',
    imageUrl: 'test-image-url',
    email: 'test@example.com'
};

// Mock-Funktionen für den Store
const mockSetProfile = vi.fn();
const mockUpdateProfile = vi.fn();
const mockClearProfile = vi.fn();

// Mock für useUserProfileStore
vi.mock('../../stores/useUserProfileStore', () => ({
    useUserProfileStore: vi.fn((selector) => selector({
        profile: mockProfile,
        isLoading: false,
        error: null,
        setProfile: mockSetProfile,
        updateProfile: mockUpdateProfile,
        clearProfile: mockClearProfile
    }))
}));

// Mock für UserProfile-Komponente
vi.mock('../atoms/UserProfile', () => ({
    default: vi.fn(({ hideText }) => (
        <div data-testid="user-profile-mock" data-hide-text={hideText ? 'true' : 'false'}>
            User Profile Mock {hideText ? '(text hidden)' : '(text visible)'}
        </div>
    ))
}));

// Mock für useMediaQuery
vi.mock('../../hooks/useMediaQuery', () => ({
    default: vi.fn().mockImplementation((query) => {
        // Simuliere die Medienabfrage basierend auf dem Query-String
        if (query === '(max-width: 480px)') {
            // Bei Tests für kleine Bildschirme wird dieser Wert in beforeEach überschrieben
            return false;
        }
        return false;
    })
}));

// Import des gemockten Hooks für explizite Kontrolle in Tests
import useMediaQuery from '../../hooks/useMediaQuery';

describe('MobileHeader', () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });

    afterEach(() => {
        // Setze den Mock zurück
        vi.mocked(useMediaQuery).mockImplementation(() => false);
        vi.restoreAllMocks();
    });

    // Unit Test - Grundlegendes Rendering
    it('should render with title and sidebar button', () => {
        const handleSidebar = vi.fn();
        render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} data-testid="mobile-header" />);

        // Prüfe ob der Titel angezeigt wird
        expect(screen.getByText('Test Title')).toBeInTheDocument();

        // Prüfe ob der Button zum Öffnen der Sidebar vorhanden ist
        const sidebarButton = screen.getByRole('button');
        expect(sidebarButton).toBeInTheDocument();
        expect(screen.getByText('Sidebar öffnen')).toBeInTheDocument();
    });

    // Unit Test - Prüft, ob der onOpenSidebar-Callback aufgerufen wird
    it('should call onOpenSidebar when sidebar button is clicked', () => {
        const handleSidebar = vi.fn();
        render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} />);

        // Finde den Button und klicke ihn
        const sidebarButton = screen.getByRole('button');
        fireEvent.click(sidebarButton);

        // Prüfe ob der Callback aufgerufen wurde
        expect(handleSidebar).toHaveBeenCalledTimes(1);
    });

    // Integration Test - Überprüft, ob die UserProfile-Komponente korrekt gerendert wird
    it('should render UserProfile component', () => {
        const handleSidebar = vi.fn();
        render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} />);

        // Prüfe ob die UserProfile-Komponente (mock) gerendert wird
        expect(screen.getByTestId('user-profile-mock')).toBeInTheDocument();
    });

    // Test für responsive Design - Text auf kleinem Bildschirm verstecken
    it('should hide text in UserProfile on small screens', () => {
        // Überschreibe den Mock für diesen Test
        vi.mocked(useMediaQuery).mockImplementation(() => true);

        const handleSidebar = vi.fn();
        render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} />);

        // Prüfe ob die UserProfile-Komponente mit hideText=true aufgerufen wurde
        const userProfileMock = screen.getByTestId('user-profile-mock');
        expect(userProfileMock).toHaveAttribute('data-hide-text', 'true');
        expect(userProfileMock).toHaveTextContent('(text hidden)');
    });

    // Test für responsive Design - Text auf größerem Bildschirm anzeigen
    it('should show text in UserProfile on larger screens', () => {
        // Überschreibe den Mock für diesen Test
        vi.mocked(useMediaQuery).mockImplementation(() => false);

        const handleSidebar = vi.fn();
        render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} />);

        // Prüfe ob die UserProfile-Komponente mit hideText=false aufgerufen wurde
        const userProfileMock = screen.getByTestId('user-profile-mock');
        expect(userProfileMock).toHaveAttribute('data-hide-text', 'false');
        expect(userProfileMock).toHaveTextContent('(text visible)');
    });

    // Test Fall: Kein Profil vorhanden
    it('should render nothing when profile is not available', () => {
        // Temporärer Mock-Override für diesen Test
        vi.mocked(useUserProfileStore).mockImplementationOnce((selector) => selector({
            profile: null,
            isLoading: false,
            error: null,
            setProfile: mockSetProfile,
            updateProfile: mockUpdateProfile,
            clearProfile: mockClearProfile
        } as MockUserProfileState));

        const handleSidebar = vi.fn();
        const { container } = render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} />);

        // Wenn kein Profil vorhanden ist, sollte nichts gerendert werden
        expect(container.firstChild).toBeNull();
    });

    // Snapshot Test
    it('should match snapshot', () => {
        const handleSidebar = vi.fn();
        const { container } = render(<MobileHeader title="Test Title" onOpenSidebar={handleSidebar} />);
        expect(container).toMatchSnapshot();
    });
}); 