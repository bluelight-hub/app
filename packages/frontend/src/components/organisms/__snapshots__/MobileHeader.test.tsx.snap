// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MobileHeader > should match snapshot 1`] = `
<div>
  <div
    class="sticky top-0 z-40 flex items-center gap-x-6 bg-white dark:bg-gray-900 px-4 py-4 shadow-sm dark:shadow-gray-800 sm:px-6 lg:hidden"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-19lec04 ant-btn-text ant-btn-color-default ant-btn-variant-text -m-2.5 p-2.5 text-gray-700 dark:text-gray-300 lg:hidden"
      type="button"
    >
      <span
        class="sr-only"
      >
        Sidebar Ã¶ffnen
      </span>
      <svg
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224,128a8,8,0,0,1-8,8H112a8,8,0,0,1,0-16H216A8,8,0,0,1,224,128ZM112,72H216a8,8,0,0,0,0-16H112a8,8,0,0,0,0,16ZM216,184H40a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM34.34,141.66a8,8,0,0,0,11.32,0l40-40a8,8,0,0,0,0-11.32l-40-40A8,8,0,0,0,34.34,61.66L68.69,96,34.34,130.34A8,8,0,0,0,34.34,141.66Z"
        />
      </svg>
    </button>
    <div
      class="flex-1 text-sm/6 font-semibold text-gray-900 dark:text-gray-100"
    >
      Test Title
    </div>
    <div>
      <div
        data-hide-text="false"
        data-testid="user-profile-mock"
      >
        User Profile Mock 
        (text visible)
      </div>
    </div>
  </div>
</div>
`;
